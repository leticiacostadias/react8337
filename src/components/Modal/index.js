import React, { Component } from 'react';
import propTypes from 'prop-types';
import './index.css';

import Widget from './../Widget';

class Modal extends Component {
  static propTypes = {
    estaAberto: propTypes.bool.isRequired,
    fechaModal: propTypes.func.isRequired
  }

  handleClick = (event) => {
    if (event.target.closest('.modal__wrapper')) {
      return;
    }

    this.props.fechaModal();
  }

  render () {
    const { estaAberto } = this.props;

    return (
      <div
        className={`modal ${estaAberto && 'modal--active'}`}
        onClick={this.handleClick}
      >
        <div className="modal__wrapper">
          <Widget>
            {this.props.children}
            </Widget>
        </div>
      </div>
    );
  }
}

export default Modal;