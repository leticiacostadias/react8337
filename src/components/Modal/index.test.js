import React from 'react';
import Modal from './index';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Modal', () => {
  it('deve renderizar filhos', () => {
    const modal = renderer.create(
      <Modal estaAberto={true} fechaModal={() => {}}>
        <h1>Título do modal</h1>
      </Modal>
    );
    const arvoreDOM = modal.toJSON();

    expect(arvoreDOM).toMatchSnapshot();
  });

  it('deve ter uma classe de estado ativo quando estaAberto for true', () => {
    const modal = shallow(
      <Modal estaAberto={true} fechaModal={() => {}}>
        <h1>Título do modal</h1>
      </Modal>
    );

    const modalOverlay = modal.find('.modal');

    expect(modalOverlay.hasClass('modal--active')).toBeTruthy();
  });

  it('não deve ter uma classe de estado ativo quando estaAberto for false', () => {
    const modal = shallow(
      <Modal estaAberto={false} fechaModal={() => {}}>
        <h1>Título do modal</h1>
      </Modal>
    );

    const modalOverlay = modal.find('.modal');

    expect(modalOverlay.hasClass('modal--active')).toBeFalsy();
  });

  it('não deve acionar fechaModal quando .modal__wrapper for clicado', () => {
    const mockFechaModal = jest.fn();
    const modal = shallow(
      <Modal estaAberto={true} fechaModal={mockFechaModal}>
        <h1>Título do modal</h1>
      </Modal>
    );

    const mockEvent = {
      target: {
        closest: () => true
      }
    };

    modal.simulate('click', mockEvent);
    expect(mockFechaModal.mock.calls.length).toBe(0);
  });

  it('deve acionar fechaModal quando .modal for clicado', () => {
    const mockFechaModal = jest.fn();
    const modal = shallow(
      <Modal estaAberto={true} fechaModal={mockFechaModal}>
        <h1>Título do modal</h1>
      </Modal>
    );

    const mockEvent = {
      target: {
        closest: () => false
      }
    };

    modal.simulate('click', mockEvent);
    // expect(mockFechaModal.mock.calls.length).toBe(1);
    expect(mockFechaModal).toHaveBeenCalled();
  });
});
