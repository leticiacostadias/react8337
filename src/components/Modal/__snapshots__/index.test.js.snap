// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal deve renderizar filhos 1`] = `
<div
  className="modal modal--active"
  onClick={[Function]}
>
  <div
    className="modal__wrapper"
  >
    <div
      className="widget"
    >
      <h1>
        TÃ­tulo do modal
      </h1>
    </div>
  </div>
</div>
`;
